src/http_workers/work_runner.rs

```diff
@@ -0,0 +1,147 @@
+use std::{sync::Arc, time::SystemTime};
+
+use async_trait::async_trait;
+use chrono::Utc;
+use hyper::{
+    client::{Builder, HttpConnector},
+    Body, Client, Method, Request,
+};
+
+use crate::{
+    cache::Cache,
+    identity::Identity,
+    twin::{SubstrateTwinDB, Twin, TwinDB},
+    types::{Message, QueuedMessage},
+    workers::Work,
+};
+
+use anyhow::{Context, Result};
+
+#[derive(Clone)]
+pub struct WorkRunner<C, I>
+where
+    C: Cache<Twin>,
+    I: Identity,
+{
+    twin_db: SubstrateTwinDB<C>,
+    identity: I,
+}
+
+impl<C, I> WorkRunner<C, I>
+where
+    C: Cache<Twin>,
+    I: Identity,
+{
+    pub fn new(twin_db: SubstrateTwinDB<C>, identity: I) -> Self {
+        Self { twin_db, identity }
+    }
+
+    async fn get_twin(&self, twin_id: usize, retires: usize) -> Option<Twin> {
+        for _ in 0..retires {
+            let twin = self.twin_db.get(twin_id.to_owned() as u32).await;
+
+            match twin {
+                Ok(twin) => {
+                    return Some(twin);
+                }
+                Err(err) => {
+                    log::debug!(
+                        "can not retrieve twin from substrate for this id '{}' because of {}",
+                        twin_id,
+                        err
+                    );
+                    continue;
+                }
+            };
+        }
+        None
+    }
+
+    fn encrypt_dat(dat: String, pubkey: String) -> Result<String> {
+        todo!()
+    }
+
+    async fn send_msg(twin: Twin, uri_path: String, msg: Message, dst: usize) -> Result<()> {
+        let req = Request::builder()
+            .method(Method::POST)
+            .header("content-type", "application/json");
+
+        let req = req.uri(format!(
+            "{}/{}",
+            twin.address.trim_end_matches('/'),
+            uri_path
+        ));
+
+        let req = req
+            .body(Body::from(serde_json::to_vec(&msg).unwrap()))
+            .context(format!(
+                "can not construct request body for this id '{}'",
+                dst
+            ))?;
+
+        let req = Request::from(req);
+        Client::new()
+            .request(req)
+            .await
+            .map(|op| ())
+            .context("failure of message delivery!")
+    }
+}
+
+#[async_trait]
+impl<C, I> Work for WorkRunner<C, I>
+where
+    C: Cache<Twin>,
+    I: Identity,
+{
+    type Job = QueuedMessage;
+
+    async fn run(&self, job: Self::Job) {
+        //identify uri and extract msg
+        let (uri_path, msg) = match job {
+            QueuedMessage::Forward(msg) => (String::from("rmb-remote"), msg),
+            QueuedMessage::Reply(msg) => (String::from("rmb-reply"), msg),
+        };
+
```

Comment:
According to the specs. if a message is failed to be delivered to a remote twin. A response is sent back to the caller 

> Quote from issue

If this is a Forward message. it means we are the sender of the request. Hence the process that initiated this call lives on the same local machine. Hence we can simply push a response back to this caller and tell him we failed. This is can be done by creating a response Message with all valid fields (specially the ret queue) and then call storage.reply() method. This will basically push your response to the caller and hence it will know about this error. The Message need src, dst, and err fields set with empty data.

----

This can be done cleaner in the loop like
```
for id in destinations {
     result = self.handle(id, msg);
     if result is error and forward message {
          create and push repl.
     }
}
```